src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -169,6 +169,229 @@ public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
+            else if (vectorMatch.Success)
+            {
+                vectorSize = int.Parse(vectorMatch.Groups[1].Value);
+                typePostfix = vectorMatch.Groups[2].Value;
+                if (typePostfix == "f") typePostfix = "";
+                typeName = "Vector" + vectorSize + typePostfix;
+            }
+            else
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            NameTable nameTable = overload.NameTable.New();
+            Parameter[] singleParameters = overload.InputParameters.ToArray();
+            Parameter[] spanParameters = overload.InputParameters.ToArray();
+            Parameter[] arrayParameters = overload.InputParameters.ToArray();
+            Parameter? countParameter = null;
+            Parameter? ptrParam = null;
+            Parameter? vectorParam = null;
+            for (var i = 0; i < singleParameters.Length; i++)
+            {
+                Parameter parameter = singleParameters[i];
+                if (parameter.Length == null) continue;
+                if (parameter.Type is CSPointer ptr && ptr.BaseType is CSType baseType)
+                {
+                    if (!_existingVectorTypes.Contains(typeName))
+                    {
+                        // Some vector methods arent actually vectors.
+                        // So we just overload them like their normal types.
+                        typeName = baseType.TypeName;
+                    }
+
+                    Constant constant;
+                    if (parameter.Length is Constant cnst)
+                    {
+                        constant = cnst;
+                    }
+                    else if (parameter.Length is BinaryOperation binary)
+                    {
+                        ParameterReference reference;
+                        if (binary.Left is ParameterReference leftRef && binary.Right is Constant rightConst)
+                        {
+                            reference = leftRef;
+                            constant = rightConst;
+                        }
+                        else if (binary.Right is ParameterReference rightRef && binary.Left is Constant leftConst)
+                        {
+                            reference = rightRef;
+                            constant = leftConst;
+                        }
+                        else continue;
+
+                        countParameter = singleParameters.First(p => p.Name == reference.ParameterName);
+                    }
+                    else continue;
+
+                    if (constant.Value == vectorSize)
+                    {
+                        CSType vector = new CSType(typeName, baseType.Constant);
+                        singleParameters[i] = parameter with
+                        {
+                            Type = new CSRef(ptr.Constant || baseType.Constant ? CSRef.Type.In : CSRef.Type.Ref,
+                                vector),
```

Comment:
Why are we checking both the pointer and base type for constant-ness? Is the spec inconsistent?

Reply:
We are also doing this expression multiple times, we can do `bool constant = ptr.Constant || baseType.Constant` and simplify these expressions.

Reply:
Its not because the spec is inconsistent really. Pretty sure just a check on baseType would be enough. But i noticed this for our ReadonlySpans too. The problem is that both things kind of mean the same thing. So as far as i have understood it we kind of need to check if any of them are true. Might have misunderstood this tho.

Reply:
In c there are slight differences between the two, although I don't think making the pointer `const` wouldn't mean to use when passing the parameter. Only the value itself being const matters.